### PR TITLE
Automatically get miner id by logs on first start

### PR DESCRIPTION
**Pull request by memedmusic**
Some people have mentioned it on discord, so i did it.
I tested it and i've found no bugs (yay)

### CODE REVIEW

**Review by vtheskeleton — CHANGES_REQUESTED** [review]
if you find them then you should ask the user if they want to use that one, if yes, cool continue to menu, if not then let them do the setup.

**Comment by memedmusic** [comment]
Cool, that should be all done

**Comment by vtheskeleton** [comment]
me when ask people questions
can you request a review next time though

**Comment by vtheskeleton** [comment]
oh wait you just did

File: src/setup.js [hunk 1]
```diff
@@ -6,6 +6,66 @@ const path = require('path');
...
+			}
+			const rigIDRegex = /^NiceHash rig ID: [a-z0-9]{15}$/m;
+			rigID = logFileContent.match(rigIDRegex);
+			if (rigID) rigID = rigID.join(" ");
+			if (rigID) rigID = rigID.split(": ")[1];
+			return rigID;
+		}
+		rigID = getIDFromLogs("main.log") ?? getIDFromLogs("main.old.log")
+		if (!rigID) {
+			run(false, true)
```
Comment (vtheskeleton) [inline src/setup.js:33]: whart?? run only takes one argument no?

Comment (vtheskeleton) [reply src/setup.js:33]: i may be blind but im not seeing any signs of actually asking for the things either

Comment (memedmusic) [reply src/setup.js:33]: Asking for things on line 36-49, and yeah run only does take 1 arg, i forgot to remove it lol

**Comment by vtheskeleton** [comment]
my brain is hurting

**Comment by vtheskeleton** [comment]
what the fuck is this spaghetti

**Comment by memedmusic** [comment]
more spaghetti = more better :D

**Comment by vtheskeleton** [comment]
kill me

**Comment by vtheskeleton** [comment]
image

**Comment by vtheskeleton** [comment]
Closing this after our discussion on discord. Still open to the idea if issues are fixed though

**Pull request closed without merging**